modules/library/jdbc/src/main/java/org/geotools/jdbc/PFSequencedPrimaryKeyColumn.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.jdbc;
+
+/**
+ * Represents a column of a primary key which has an associated pre-fetch sequence used to generate
+ * its values. Different from the SequencedPrimaryKeyColumn, the value for this column will be
+ * fetched from a sequence before the inserting in the data store.
```

Comment:
Very good comment that explains a lot, thank you for this.

Reply:
thanks